crates/recursor/src/recursor.rs

```diff
@@ -59,17 +67,12 @@ impl RecursorBuilder {
     /// # Panics
     ///
     /// This will panic if the roots are empty.
-    pub fn build(&self, roots: impl Into<NameServerConfigGroup>) -> Recursor {
-        #[cfg(not(feature = "dnssec"))]
-        let security_aware = false;
-        #[cfg(feature = "dnssec")]
-        let security_aware = self.security_aware;
-
+    pub fn build(&self, roots: impl Into<NameServerConfigGroup>) -> Result<Recursor, Error> {
```

Comment:
If we're going to flip-flop on the fallibility of this method, let's just keep it fallible?